File: mobile/android/air/SubModules/WalletCore/src/main/java/org/mytonwallet/app_air/walletcore/stores/ConfigStore.kt
package org.mytonwallet.app_air.walletcore.stores

object ConfigStore {
    var isCopyStorageEnabled: Boolean? = null
    var supportAccountsCount: Double? = null
    var isLimited: Boolean? = null
    var countryCode: String? = null
    var isAppUpdateRequired: Boolean? = null

    fun init(configMap: Map<String, Any>?) {
        if (configMap == null) return
        isCopyStorageEnabled = configMap["isCopyStorageEnabled"] as? Boolean
        supportAccountsCount = configMap["supportAccountsCount"] as? Double
        isLimited = configMap["isLimited"] as? Boolean
        countryCode = configMap["countryCode"] as? String
        isAppUpdateRequired = configMap["isAppUpdateRequired"] as? Boolean
    }
}

File: mobile/ios/Air/SubModules/WalletCore/Stores/ConfigStore.swift
//
//  ConfigStore.swift
//  MyTonWalletAir
//
//  Created by Sina on 11/6/24.
//

import Foundation
public class ConfigStore {
    
    public struct Config {
        let isCopyStorageEnabled: Bool?
        let supportAccountsCount: Bool?
        let isLimited: Bool?
        public let countryCode: String?
        
        public init(dictionary: [String: Any]) {
            self.isCopyStorageEnabled = dictionary["isCopyStorageEnabled"] as? Bool
            self.supportAccountsCount = dictionary["supportAccountsCount"] as? Bool
            self.isLimited = dictionary["isLimited"] as? Bool
            self.countryCode = dictionary["countryCode"] as? String
        }
        
    }
    
    private init() {}
    
    private static let queue = DispatchQueue(label: "org.mytonwallet.app.config_store", attributes: .concurrent)
    
    private static var _config: Config? = nil
    public internal(set) static var config: Config? {
        get {
            return queue.sync { _config }
        }
        set {
            queue.async(flags: .barrier) { _config = newValue }
        }
    }
    
    public static func clean() {
        queue.async(flags: .barrier) {
            _config = nil
        }
    }
}
